### ISSUE TITLE
[BUG] avoid-dynamic False Positive

### ISSUE THREAD

**Issue by g5becks** [dart-code-checker/dart-code-metrics#1234]
### Environment and configuration

DCM version: dart_code_metrics: ^5.7.4
Dart SDK version: 3.0.0 (stable)

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  extends:
    - package:dart_code_metrics_presets/all.yaml
  metrics-exclude:
    - test/**
  rules:
    - avoid-late-keyword: false
    - prefer-static-class: false
    - no-magic-number: false
    - format-comment: false
```
</details>

### What did you do?

```dart

import 'package:rxdart/rxdart.dart' show ReplaySubject;
import 'package:technical_indicators/src/types.dart';

import 'quotes.dart';
import 'util.dart';

Decimal _d(String s) => Decimal.parse(s);

typedef Series<T> = Stream<T>;

typedef QuoteSeries = ReplaySubject<Quote>;
```

### What did you expect to happen?

No dynamic types were used, so I expect the warning not to appear.

### What actually happened?

Entire codebase is giving the waring for avoid-dynamic based on this one usage , even though a dynamic type was never used.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by incendial** [comment]
@g5becks do you have the same problem if you use any Dart version below 3.0?

**Comment by g5becks** [comment]
> @g5becks do you have the same problem if you use any Dart version below 3.0?

I actually haven't tried that. I will give a test and report back.
